src/basic/architecture.h

```diff
@@ -48,7 +49,7 @@ typedef enum {
         _ARCHITECTURE_INVALID = -EINVAL,
```

Comment:
Maybe, it is better to include -ERRNO_MAX?

Reply:
the functions returning `Architecture` cannot fail actually, right now. (I mean only architecture_to_string() can, but only via the `_ARCHITECTURE_INVALID`/`-EINVAL` special case). Hence, following our usual rule of adding a `-ERRNO_MAX` entry only where we actually need it I didn't add it here.